dill/_dill.py

```diff
@@ -876,12 +939,100 @@ def __getattribute__(self, attr):
             attrs[index] = ".".join([attrs[index], attr])
         return type(self)(attrs, index)
 
+
+
+# Used to stay compatible with versions of dill whose _create_cell functions
+# do not have a default value.
+# Can be safely replaced removed entirely (replaced by empty tuples for calls to
+# _create_cell) once breaking changes are allowed.
+_CELL_EMPTY = sentinel('_CELL_EMPTY')
+_CELL_EMPTY_shim = BuiltinShim('_CELL_EMPTY', None)
```

Comment:
It seems like all you really need is `_CELL_EMPTY` and `_CELL_REF`.  Using `_CELL_EMPTY_shim` adds a `__call__` to a constant, which is a bit weird.  Also, it would seem that `_CELL_EMPTY` is always defined in the line above.

Reply:
It is not defined in previous versions of `dill`. When a function with an uninitialized value is pickled in 0.3.5 and unpickled in 0.3.4, the default value of None allows for the function to be unpickled in Python 3.7+. `_CELL_EMPTY` really is only needed in PyPy 2.7.

`_CELL_REF` can be deleted entirely when a breaking change to the codebase needs to be made. I still left it in as `None` instead of removing it because `_create_cell` didn't have a default value in 0.3.4. It is just left as a hint that the argument to `_create_cell` can be removed in that case and doesn't have to stay `None` in future versions.

Reply:
Ah... that's right.  That's why you are using a lazy definition of the shim.  Sorry, my oversight.  I get why you are doing this.  Any thoughts on any issues how you re using `_CELL_EMPTY_shim` which adds a `__call__` to a constant, which is a bit unusual.

Reply:
I'll make it so that `self.__call__` is deleted if the alternative is not callable. That will probably make more sense.